rmw_connext_xtypes_dynamic_cpp/src/functions.cpp

```diff
@@ -889,16 +1095,76 @@ rmw_take_response(
   const rmw_client_t * client,
   void * ros_request_header, void * ros_response, bool * taken)
 {
-  *taken = false;
-  return RMW_RET_ERROR;
+  if (taken == NULL) {
+    rmw_set_error_string("taken argument can't be null");
+    return RMW_RET_ERROR;
+  }
+
+  if (client->implementation_identifier != rti_connext_dynamic_identifier) {
+    rmw_set_error_string("client handle not from this implementation");
+    // printf("but from: %s\n", client->implementation_identifier);
+    return RMW_RET_ERROR;
+  }
+
+  ConnextDynamicClientInfo * client_info = \
+    static_cast<ConnextDynamicClientInfo *>(client->data);
+  connext::Requester<DDS_DynamicData, DDS_DynamicData> * requester = client_info->requester_;
+
+  connext::LoanedSamples<DDS::DynamicData> replies = requester->take_replies(1);
+  typedef connext::LoanedSamples<DDS::DynamicData>::iterator iterator;
+  for (iterator it = replies.begin(); it != replies.end(); ++it) {
+    printf("  rmw_take_response() got a reply\n");
+  }
+  if (replies.begin() != replies.end() && replies.begin()->info().valid_data) {
+    _take(&replies.begin()->data(), ros_response, client_info->response_members_);
+
+    rmw_request_id_t & req_id = *(reinterpret_cast<rmw_request_id_t *>(ros_request_header));
+    req_id.sequence_number = ((int64_t)replies.begin()->related_identity().sequence_number.high)
+      << 32 | replies.begin()->related_identity().sequence_number.low;
```

Comment:
The use of parenthesis here seems strange to me. I would have thought that you would have included the `<< 32` in the parenthesis. Was this intentional?